frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java

```diff
@@ -123,6 +139,21 @@ private boolean isBranchNameRepeated(String fileContent, String branchName) {
     return lines.stream().filter(line -> line.equals(branchName)).count() > 1;
   }
 
+  private String getParentBranchName(PsiFile file, String branchName) throws BranchLayoutException {
+    val branchLayoutReader = RuntimeBinding.instantiateSoleImplementingClass(IBranchLayoutReader.class);
+    val branchLayout = branchLayoutReader.read(Path.of(file.getVirtualFile().getPath()));
+    BranchLayoutEntry parentEntry;
+    try {
+      parentEntry = Objects.requireNonNull(branchLayout.getEntryByName(branchName)).getParent();
+    } catch (NullPointerException e) { // might appear if saveDocumentBeforeCheck(file) has not completed yet
+      return "";
```

Comment:
This is a workaround for scenarios where `saveDocumentBeforeCheck(file)` has not completed before `getParentBranchName` is invoked. In such case we just return `""` which means that _checked branch has no parent_. In reality - new branch entry might have a parent, so returning `""` does not seem right. However - in practice, when `saveDocumentBeforeCheck(file)` is completed, then whole `MacheteAnnotator#annotate` gets invoked again... which results in valid result with proper parent branch name.

Reply:
So, it saves us from `NullPointerException` and **does not** break the logic in practice.

Reply:
Same workaround for `BranchLayoutException` in the same method...